javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/instrumentation/indy/IndyTypeTransformerImpl.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.tooling.instrumentation.indy;
+
+import io.opentelemetry.javaagent.extension.instrumentation.InstrumentationModule;
+import io.opentelemetry.javaagent.extension.instrumentation.TypeTransformer;
+import io.opentelemetry.javaagent.tooling.bytebuddy.ExceptionHandlers;
+import net.bytebuddy.agent.builder.AgentBuilder;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+
+public final class IndyTypeTransformerImpl implements TypeTransformer {
+  private AgentBuilder.Identified.Extendable agentBuilder;
+
+  private final InstrumentationModule instrumentationModule;
+
+  public IndyTypeTransformerImpl(
+      AgentBuilder.Identified.Extendable agentBuilder, InstrumentationModule module) {
+    this.agentBuilder = agentBuilder;
+    this.instrumentationModule = module;
+  }
+
+  @Override
+  public void applyAdviceToMethod(
+      ElementMatcher<? super MethodDescription> methodMatcher, String adviceClassName) {
+    Advice.WithCustomMapping withCustomMapping =
+        Advice.withCustomMapping()
+            .with(new Advice.AssignReturned.Factory().withSuppressed(ClassCastException.class))
```

Comment:
This was taken directly from the implementation in the elastic-apm-agent. I'm not the original author of that code but I assume the reasoning is the following:
Imagine you are writing an Advice which wants to assign to a member variable of the instrumented class with type `Foo`. So you simply return a `Foo` instance from your advice, which gets assigned to the member variable.

Now your instrumented library changes: The type of the member variable is changed from `Foo` to the unrelated `Bar`. Your advice will now trigger a `ClassCastException` after the advice has executed because bytebuddy inlines code to assign the result of your advice to the member variable (which is not assignable).

To my understanding the `.withSuppressed` ensures that such exceptions are caught and handled just like exceptions which originate from the execution of the advice code. 

However, after thinking about it I don't see why we would limit this to `ClassCastExceptions`. For example when unpacking from an array, `NullPointerException`s could maybe occur if the array is `null`.

I'd suggest that we change this to `.withSuppressed(Throwable.class)` to be on the safe side.

Reply:
will this use the exception handler from https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/44e908e52cd9efa00b4df897ed95c95cc66b04ed/javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/instrumentation/TypeTransformerImpl.java#L32 or just suppress it?